src/coverlet.core/Instrumentation/Instrumenter.cs

```diff
@@ -447,6 +447,14 @@ private void InstrumentIL(MethodDefinition method)
                 var sequencePoint = method.DebugInformation.GetSequencePoint(instruction);
                 var targetedBranchPoints = branchPoints.Where(p => p.EndOffset == instruction.Offset);
 
+                // Check if the instruction is coverable
+                if (CecilSymbolHelper.IsMoveNextInsideAsyncStateMachine(method) &&
+                    CecilSymbolHelper.IsNotCoverableInstructionAfterExceptionThrown(instruction))
```

Comment:
Can you fold this two method in one and hide `IsMoveNextInsideAsyncStateMachine` that is used inside `CecilSymbolHelper`.
Something like `SkipInstructionAfterThrownInsideMoveNextAsyncStateMachine`